src/modules/APM.ts

```diff
@@ -62,7 +62,8 @@ export const startExecutionTrace = async (name: string): Promise<Trace> => {
   const id = await NativeAPM.startExecutionTrace(name, timestamp);
 
   if (!id) {
-    throw new Error(TRACE_NOT_STARTED_APM_NOT_ENABLED);
+    console.error(TRACE_NOT_STARTED_APM_NOT_ENABLED);
+    return Promise.reject(TRACE_NOT_STARTED_APM_NOT_ENABLED);
```

Comment:
Rejecting the promise still results in the same behavior of crashing the function calling `startExecutionTrace`.

I think the proper fix would be to return `Promise<Trace | undefined>` but that would be a breaking change as it'd change the API signature.

Since it's already deprecated and is going to be removed in the future, I think we could leave it unchanged until removed and ask users to use `startFlow` instead.

What do you think? We can discuss this with the team further.